### PR TITLE
Add a new compiler object method: has_members

### PR DESCRIPTION
**Pull request by nirbheek**
Identical to `cc.has_member`, except that this takes multiple members and all of them must exist else it returns false.

This is useful when you want to verify that a structure has all of a given set of fields. Individually checking each member is horrifying.

### CODE REVIEW

File: mesonbuild/interpreter.py [hunk 1]
```diff
@@ -676,6 +677,28 @@ def has_member_method(self, args, kwargs):
                  '" has member "', mlog.bold(membername), '": ', hadtxt, sep='')
         return had
 
+    def has_members_method(self, args, kwargs):
+        check_stringlist(args)
+        typename = args[0]
+        membernames = args[1:]
+        prefix = kwargs.get('prefix', '')
+        if not isinstance(prefix, str):
+            raise InterpreterException('Prefix argument of has_function must be a string.')
```
Comment (QuLogic) [inline mesonbuild/interpreter.py]: `has_members`

**Comment by nirbheek** [comment]
The test failure seems to be because of Python 3.4. I'll fix it by not doing that.

**Comment by jpakkane** [comment]
To reduce duplication, `has_member` should be implemented by having it call `has_members`.

**Comment by nirbheek** [comment]
I've done that for the `compiler.py` bits, but it seemed useless for the `interpreter.py` bits.

**Comment by nirbheek** [comment]
Btw, I think we should do argument checking and transformation (`check_stringlist`, `stringlistify`, etc) using decorators. It's much clearer IMO.

**Pull request merged**